addons/website_sale_wallet/models/wallet.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, api, _
+
+
+class Wallet(models.Model):
+    _inherit = ['website.multi.mixin', 'wallet']
+    _name = "wallet"
+
+
+class WalletTransaction(models.Model):
+    _inherit = "wallet.transaction"
+
+    @api.constrains("gift_card_id")
+    def _link_gift_card(self):
+        super(WalletTransaction, self)._link_gift_card()
+        for record in self:
+            record.gift_card_id.website_id = record.wallet_id.website_id
```

Comment:
look like it is a related ...

Reply:
yes it is.
But the user can specify the website when purchasing, or manually